src/dump.c

```diff
@@ -257,16 +259,16 @@ static int type_recursively_external(jl_datatype_t *dt);
 
 static int type_parameter_recursively_external(jl_value_t *p0) JL_NOTSAFEPOINT
 {
-    jl_datatype_t *p = (jl_datatype_t*)p0;
-    while (jl_is_unionall(p)) {
-        if (!type_parameter_recursively_external(((jl_unionall_t*)p)->var->lb))
-            return 0;
-        if (!type_parameter_recursively_external(((jl_unionall_t*)p)->var->ub))
-            return 0;
-        p = (jl_datatype_t*)((jl_unionall_t*)p)->body;
-    }
-    if (!jl_is_datatype(p) || p->uid == 0)
+    if (!jl_is_concrete_type(p0))
         return 0;
+    jl_datatype_t *p = (jl_datatype_t*)p0;
+    //while (jl_is_unionall(p)) {
+    //    if (!type_parameter_recursively_external(((jl_unionall_t*)p)->var->lb))
+    //        return 0;
+    //    if (!type_parameter_recursively_external(((jl_unionall_t*)p)->var->ub))
+    //        return 0;
+    //    p = (jl_datatype_t*)((jl_unionall_t*)p)->body;
+    //}
```

Comment:
delete?

Reply:
It seemed probably good, but wasn't really functional anymore, and we're not to the point yet where this is a performance bottleneck.